lib/index.js

```diff
@@ -309,14 +309,23 @@ Mongoose.prototype.disconnect.$hasSideEffects = true;
  *     var collectionName = 'actor'
  *     var M = mongoose.model('Actor', schema, collectionName)
  *
- * @param {String} name model name
+ * @param {String|Function} name model name or class extending Model
  * @param {Schema} [schema]
- * @param {String} [collection] name (optional, induced from model name)
+ * @param {String} [collection] name (optional, inferred from model name)
  * @param {Boolean} [skipInit] whether to skip initialization (defaults to false)
  * @api public
  */
 
 Mongoose.prototype.model = function(name, schema, collection, skipInit) {
+  var model;
+  if (typeof name === 'function') {
+    model = name;
+    name = model.name;
+    if (!(model.prototype instanceof Model)) {
```

Comment:
Correct me if I'm wrong.. I don't think this would work in the case when using a Base class on an individual class.

```javascript
export class Foo extends MyBase {...}
```